src/sentry/utils/snuba.py

```diff
@@ -218,9 +221,31 @@ def get_project_issues(project_ids, issue_ids=None):
         hashes = GroupHash.objects.filter(group_id__in=issue_ids)
     else:
         hashes = GroupHash.objects.filter(project__in=project_ids)
+
+    hashes = list(hashes)
+    if not hashes:
+        return []
+
+    hashes_by_project = {}
+    for h in hashes:
+        hashes_by_project.setdefault(h.project_id, []).append(h.hash)
+
+    tombstones = GroupHashTombstone.objects.filter(
```

Comment:
Is is possible to get these with the hashes in a single django query with `select_related`?

Reply:
There's no FK relation here, so that isn't possible is it?

Reply:
I guess I could do raw SQL?

Reply:
oh oops. didn't realize that. Next question then, should GroupHashTombstone have a FK to GroupHash?

Reply:
When a Group is deleted the GroupHashes are deleted with it (and GroupHashTombstones are created).

In theory we could maybe change that. The tombstone date could be a column on the GroupHash itself, and then we wouldn't delete the GroupHash rows at all? There's a lot of existing code and implications around deleting the rows (freeing them up to be recreated in a new group, or used in an existing one). I went for the new model because this is kind of a Snuba-specific hack. Maybe @tkaemming has an opinion here though?

Reply:
Yeah I guess in a perfect world its just a column on GroupHash, but I understand now why it cant be a FK. Just took me a minute :) I think this is fine then.

Reply:
> Maybe @tkaemming has an opinion here though?

Not a strong one, I think it could be done either way reasonably easily. I do think we'd want to consider moving it into Snuba itself if we ever can figure out how to replicate the hash data from Postgres into CH, but I'm not sure a hypothetical future improvement should really dictate the design here.